### ISSUE TITLE
Game crashed

### ISSUE THREAD

**Issue by SankarGanesh82** [mopub/mopub-android-sdk#468]
Caused by java.lang.NullPointerException: Attempt to invoke interface method 'void com.mopub.mobileads.AdLifecycleListener$LoadListener.onAdLoaded()' on a null object reference
at com.mopub.mobileads.MoPubFullscreen$1.onResponse(MoPubFullscreen.java:260)
at com.mopub.volley.toolbox.ImageLoader$4.run(ImageLoader.java:504)
at android.os.Handler.handleCallback(Handler.java:938)
at android.os.Handler.dispatchMessage(Handler.java:99)
at android.os.Looper.loop(Looper.java:223)
at android.app.ActivityThread.main(ActivityThread.java:7656)
at java.lang.reflect.Method.invoke(Method.java)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)

#### Steps to reproduce the behavior:
Please list all relevant steps to reproduce the observed behavior.

#### Expected behavior:
As concisely as possible, describe the expected behavior.

#### Observed behavior:
As concisely as possible, describe the observed behavior.

#### Evidence:
Device log files, Network log file, etc.

**Comment by chauduyphanvu** [comment]
@SankarGanesh82, thanks for the report. This crash should have been fixed in v5.14.0. Do you happen to have a Unity project that reproduces the issue that we can check?

Vu Chau
MoPub Team

**Comment by SankarGanesh82** [comment]
We are using **_latest Mopub Unity SDK version 5.16.4 and Unity LTS version 2019.4.16f1_** and found two different crashes in Google Play Console. Please give us solution ASAP.

java.lang.Error: FATAL EXCEPTION [main]
Unity version     : 2019.4.16f1
Device model      : Xiaomi M2004J19PI
Device fingerprint: POCO/shiva/shiva:10/QP1A.190711.020/V12.0.3.0.QJRINXM:user/release-keys
Caused by
  at com.mopub.mobileads.MoPubFullscreen$1.onResponse (MoPubFullscreen.java:260)
  at com.mopub.volley.toolbox.ImageLoader$4.run (ImageLoader.java:504)
  at android.os.Handler.handleCallback (Handler.java:883)
  at android.os.Handler.dispatchMessage (Handler.java:100)
  at android.os.Looper.loop (Looper.java:224)
  at android.app.ActivityThread.main (ActivityThread.java:7561)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:539)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:995)


java.lang.Error: FATAL EXCEPTION [main]
Unity version     : 2019.4.16f1
Device model      : OPPO CPH2179
Device fingerprint: OPPO/CPH2179/OP4F2F:10/QP1A.190711.020/1612405462:user/release-keys
Caused by
  at com.mopub.mobileads.MoPubFullscreen$1.onErrorResponse (MoPubFullscreen.java:266)
  at com.mopub.volley.toolbox.ImageLoader$4.run (ImageLoader.java:506)
  at android.os.Handler.handleCallback (Handler.java:883)
  at android.os.Handler.dispatchMessage (Handler.java:100)
  at android.os.Looper.loop (Looper.java:237)
  at android.app.ActivityThread.main (ActivityThread.java:7824)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:512)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1027)

<img width="1242" alt="Screenshot 2021-04-13 at 7 35 45 PM" src="https://user-images.githubusercontent.com/55496751/114566002-77da8c80-9c8f-11eb-92a5-f6d1b7346138.png">
<img width="1245" alt="Screenshot 2021-04-13 at 7 33 33 PM" src="https://user-images.githubusercontent.com/55496751/114565749-33e78780-9c8f-11eb-9de7-c09aad2fc7a3.png">

**Comment by chauduyphanvu** [comment]
@SankarGanesh82, from the stack trace it looks like the ad listener is null, causing the crashes. Could you tell us a bit more about how you are requesting and serving ads in your app?

Let me relay this to the rest of the team and we'll see what we can do.

**Comment by SankarGanesh82** [comment]
We are following as per Mopub's document. While app launch we will be checking the consent and after then initiating Mopub. Subsequently we will request for banner, interstitial and rewarded with respective ad unit ids. Before initiating Mopub all the ad listeners will be registered.

We are using Mopub so longtime and for earlier versions we didn't get crashes in Google Play Console. We have updated Mopub Unity SDK from V5.16.0 to V5.16.4. After then only we got this issue.

**Comment by chauduyphanvu** [comment]
@SankarGanesh82, thank you for the confirmation. I've noted the information for the rest of the team.

**Comment by DFC005** [comment]
We're getting the same crash after recently upgrading to 5.16.4.

```
Caused by java.lang.NullPointerException: Attempt to invoke interface method 'void com.mopub.mobileads.AdLifecycleListener$LoadListener.onAdLoadFailed(com.mopub.mobileads.MoPubErrorCode)' on a null object reference
       at com.mopub.mobileads.MoPubFullscreen$1.onErrorResponse(MoPubFullscreen.java:266)
       at com.mopub.volley.toolbox.ImageLoader$4.run(ImageLoader.java:506)
       at android.os.Handler.handleCallback(Handler.java:914)
       at android.os.Handler.dispatchMessage(Handler.java:100)
       at android.os.Looper.loop(Looper.java:225)
       at android.app.ActivityThread.main(ActivityThread.java:7563)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:539)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:994)
```

**Comment by chauduyphanvu** [comment]
@DFC005, we have addressed variations of this crash internally for the upcoming MoPub SDK release. I'll update this thread when the SDK is in production. Thanks!

**Comment by nickgrahamuk** [comment]
@chauduyphanvu Is this crash addressed in version 5.17.0? It's not mentioned in the changelog.

**Comment by chauduyphanvu** [comment]
Hi all, these crashes have been addressed in [v5.17.0](https://github.com/mopub/mopub-android-sdk/releases/tag/v5.17.0) of the MoPub Android SDK. Check it out and let us know if you still see the issues!
